# 7.3.3自定义异常 - 正负数

#### 任务描述

本关的任务是通过用户自定义异常的机制，判断一个数是否合法，我们定义为若该数为正数，则合法；反之则不合法。

#### 相关知识

##### 自定义异常

在`Java`中，除了由`Java`系统监测到的异常(下标越界，被0除等)，用户也可以自定义异常。

用户定义的异常同样要用`try{} catch{}`捕获，但必须由用户自己抛出 `throw new MyException`。

异常是一个类,用户定义的异常必须继承自`Throwable`或`Exception`类，建议用`Exception`类。

语法结构如下所示：

```
class MyException extends Exception {    …}
```

一般的，我们会这么定义一个异常：

```
class MyException extends Exception {    public MyException(String m) {        super(m);    }}
```

上述代码中的构造方法`MyException(String m)`中通过`super()`调用了父类的构造方法，效果是输出以下结果：

```
MyException: 字符串m中的内容
```

在程序中，其抛出异常的语法为：

```
throw new MyException("");  //""为字符串m中的内容，由用户自定义
```

#### 编程要求

本关的编程要求是补全右侧代码片段中`Begin`至`End`中间的代码，具体要求如下：

1. `num`变量为命令行输入的`int`型数据，若`num`为正数，输出`The number you entered is：` + `num`；反之捕获该异常。

#### 测试说明

平台将自动编译补全后的代码，并生成若干组测试数据，接着根据程序的输出判断程序是否正确。

以下是测试样例：

样例输入：`-60` 样例输出：`chapter8.step3.MyException: Number cannot be negative!`

样例输入：`60` 样例输出：`The number you entered is: 60`

------

如果想要出人头地，首先就要耐得住寂寞，因为成功的钥匙往往就藏在寂寞的口袋里。

开始你的任务吧，祝你成功！